server/src/main/java/org/elasticsearch/indices/recovery/RecoveryTranslogOperationsRequest.java

```diff
@@ -39,18 +40,26 @@
     private int totalTranslogOps = RecoveryState.Translog.UNKNOWN;
     private long maxSeenAutoIdTimestampOnPrimary;
     private long maxSeqNoOfUpdatesOrDeletesOnPrimary;
+    private RetentionLeases retentionLeases;
```

Comment:
I think we need to initialize the retentionLeases with an empty leases in a mixed cluster?

Reply:
Great catch. I pushed 4d9fa1acddfe0b6e0ab8a91c10866ccbfa530e17.